scala-thoughts.md

```diff
@@ -0,0 +1,101 @@
+
+# Scala Thoughts
+
+### Instructions
+
+After completing the exercises, take about 20 minutes to collect and write
+about your initial impressions of Scala.  Write about them in
+scala-thoughts.md Some possible topics include:
+
+---
+
+### Questions
+
+#### 1. What's easy to do in Scala? What's not?
+
+It is easy to write code that feels like python's functional programming.
+It was very easy to chain methods and compose different chunks of code.
+It was a little harder / less elegant to write imperitive code, but definitely
+still possible.
+
+From reading the scala documentation intro, it became clear that support for
+DSLs is a key feature, under the extensibility section. This can be seein in
+the ability to customize the syntax for specific domains, the freedom of
+function call syntax (infix or postfix, curried vs uncurried,
+and zero argument vs no argument).
+
+---
+
+#### 2. What is/are your favorite language design choice(s) that the designers of
+Scala made? Why?
+
+I really like how Scala's language designers did not seem to be forcing their
+preferred paradigm upon users.  For example, the use of a functional style,
+while well supported, is not the only way to write ones code.
+
+This was encouraging, especially comparing it to Haskell, which is very
```

Comment:
I've never programmed in Haskell, but I have programmed in SML (which I hear is similar) and this is definitely the case. It made SML quite hard to learn at first.